### PR TITLE
Feature/cmaq54 beiming

### PR DESCRIPTION
**Pull request by btang1**
## PR Checklist
- [ ] This PR has been tested on an RDHPCS machine and/or WCOSS2. Please select below:
  - [ ] RDHPCS.
  - [ ] WCOSS2.

- [ ] This PR has been tested with the ufs-srweather-app workflow online-cmaq branch.

- [ ] New or updated input data is required by this PR. 

- [ ] Baselines are expected to change.

### CODE REVIEW

**Comment by btang1** [comment]
the only big change is phot.F file. Changes in other 3 files are tiny, only used to let phot.f compile.

The phot.F has been compiled successfully.

I compare EPA_CMAQ5.2_production brunch vs. NOAA_CMAQ5.2_develop_brunch. then adding those changes to EPA_CMAQ5.4 to generate 'NOAA_CMAQ5.4', which is shown here.

File: src/model/src/phot.F [hunk 1]
```diff
@@ -1193,127 +1193,135 @@ END SUBROUTINE CONVCLD_PROP_ACM
...
 !Integrate to get best attenuation value to use within canopy
-                                RJ_CORR( COL,ROW ) = IntegrateTrapezoid(ZCANX(COUNTCAN:1:-1),RJ_CORRX(COUNTCAN:1:-1)) /
+                                RJ_CORR( COL,ROW ) =IntegrateTrapezoid(ZCANX(COUNTCAN:1:-1),RJ_CORRX(COUNTCAN:1:-1)) /
      &                                               ZFL
-!                                WRITE(*,*) 'RJ_CORRX      = ', RJ_CORRX(COUNTCAN:1:-1),
-!     &                                     'ZCANX         = ', ZCANX(COUNTCAN:1:-1),
-!     &                                     'RJ_CORR (int) = ', RJ_CORR( COL,ROW )
 !Apply attenuation factors above and below canopy
-                                RJ( COL,ROW, 1, : ) = RJ( COL,ROW, 1, : )*RJ_CORR( COL,ROW )
+                                RJ( COL,ROW, LEV, : ) = RJ(COL,ROW, LEV, : )*RJ_CORR( COL,ROW )
```
Comment (ytangnoaa) [inline src/model/src/phot.F:1319]: The loop variable "LEV" covers all model layers

DO LEV = 1, NLAYS

You should not apply the canopy attenuation to all model layers

File: src/model/src/phot.F [hunk 2]
```diff
@@ -1193,127 +1193,135 @@ END SUBROUTINE CONVCLD_PROP_ACM
...
-!                                WRITE(*,*) 'LAIE      = ', Met_Data%LAIE( COL,ROW ) ,
-!     &                                     'FCH       = ', Met_Data%FCH( COL,ROW ),
-!     &                                     'FRT       = ', Met_Data%FRT( COL,ROW),
-!     &                                     'POPU      = ', Met_Data%POPU( COL,ROW),
-!     &                                     'CLU       = ', Met_Data%CLU( COL,ROW)
+                      IF ( CANOPY_SHADE ) THEN ! compute canopy shade reduction factor (RJ_CORR)
 
-                            !conditions for grid cells that do NOT have
-                            !a continuous forest canopy
+                        DO LEV = 1, NLAYS !loop through model layers
```
Comment (ytangnoaa) [inline src/model/src/phot.F:1204]: Canopy effect should be only applied to lowest 1 or 2 layers, not the whole model layers (NLAYS)

File: aqm_files.cmake [hunk 3]
```diff
@@ -17,7 +17,7 @@ list(APPEND aqm_shr_files
     src/shr/aqm_logger_mod.F90
     src/shr/aqm_model_mod.F90
     src/shr/aqm_rc_mod.F90
-    src/shr/aqm_methods.F90
+   # src/shr/aqm_methods.F90
```
Comment (ytangnoaa) [inline aqm_files.cmake:20]: Why comment out the file "src/shr/aqm_methods.F90"?

**Comment by drnimbusrain** [comment]
Hi Youhua and Beiming,

As discussed,  let's leave the NLAYS as is for now in canopy phot.F.  It
adds little to no computation. We will refine later if need be.

Thank you!!

File: src/model/src/phot.F (hunk 1 above)
Comment (btang1) [reply src/model/src/phot.F:1319]: we discussed in meeting today, using LEV is correct for now

File: aqm_files.cmake (hunk 3 above)
Comment (btang1) [reply aqm_files.cmake:20]: have not compiled aqm_methods for now. So comment it now. I believe I need to change something in this code before  I can compile it.

File: src/model/src/phot.F (hunk 2 above)
Comment (btang1) [reply src/model/src/phot.F:1204]: we discussed today that let use it for now, loop through all the layers.

**Comment by drnimbusrain** [comment]
@lwcugb  @btang1 Can we close this PR?  Seems like we have everything included now in the feature/cmaq54 since most recent merge of @lwcugb PR?

**Comment by btang1** [comment]
Yes!

Sent from my iPhone

On Oct 21, 2023, at 9:26 AM, Patrick Campbell ***@***.***> wrote:

﻿

@lwcugb<https://github.com/lwcugb> @btang1<https://github.com/btang1> Can we close this PR? Seems like we have everything included now in the feature/cmaq54 since most recent merge of @lwcugb<https://github.com/lwcugb> PR?

**Pull request closed without merging**